common/interpreter/wawaka_wasm/WawakaInterpreter.cpp

```diff
@@ -152,36 +150,41 @@ int32 WawakaInterpreter::initialize_contract(
 {
     uint8_t* buffer;
     wasm_function_inst_t wasm_func = NULL;
+    int32 result = 0;
 
     SAFE_LOG(PDO_LOG_DEBUG, "wasm initialize_contract");
     wasm_func = wasm_runtime_lookup_function(wasm_module_inst, "_ww_initialize", "(i32)i32");
     pe::ThrowIfNull(wasm_func, "Unable to locate the initialize function");
 
-    // might need to add a null terminator
-    uint32 argv[1], buf_offset;
-    argv[0] = buf_offset = (int32)wasm_runtime_module_malloc(wasm_module_inst, env.length() + 1, (void**)&buffer);
-    if (argv[0] == 0) {
-        SAFE_LOG(PDO_LOG_ERROR, "module malloc failed for some reason");
-        return 0;
-    }
-
-    memcpy(buffer, env.c_str(), env.length());
-    buffer[env.length()] = '\0';
+    uint32 argv[1], buf_offset = 0;
+    try {
+        // might need to add a null terminator
+        argv[0] = buf_offset = (int32)wasm_runtime_module_malloc(wasm_module_inst, env.length() + 1, (void**)&buffer);
+        pe::ThrowIf<pe::RuntimeError>(argv[0] == 0,
+                                      "module malloc failed for some reason");
 
-    if (! wasm_runtime_call_wasm(wasm_exec_env, wasm_func, 1, argv)) {
-        SAFE_LOG(PDO_LOG_ERROR, "execution failed for some reason");
+        memcpy(buffer, env.c_str(), env.length());
+        buffer[env.length()] = '\0';
 
+        pe::ThrowIf<pe::RuntimeError>(
+           !wasm_runtime_call_wasm(wasm_exec_env,
+                                   wasm_func, 1, argv),
+           "execution failed for some reason");
+        
+        SAFE_LOG(PDO_LOG_DEBUG, "RESULT=%u", argv[0]);
+        result = argv[0];
+    }
+    catch (pe::RuntimeError& e) {
+        SAFE_LOG(PDO_LOG_ERROR, "Exception: %s", e.what());
         const char *exception = wasm_runtime_get_exception(wasm_module_inst);
         if (exception != NULL)
-            SAFE_LOG(PDO_LOG_ERROR, "exception=%s", exception);
-
-        return 0;
+            SAFE_LOG(PDO_LOG_ERROR, "wasm exception = %s", exception);
```

Comment:
`SAFE_LOG` is only defined when `PDO_DEBUG_BUILD` is defined, so it should be placed between `{}` -- otherwise the next line goes into the `if` block.

Reply:
Good point, thanks!

Reply:
We should make sure we avoid this issue in our other code, too, then :)

Reply:
OTOH, the lone semicolon at the end of the statement remains with an empty `SAFE_LOG`, so this issue is already addressed.

Reply:
technically! though I thought we had a policy to always use brackets.

Reply:
i'm actually surprised that clang-format didn't enforce that ...